src/common/Permission/CrudPermission.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useEffect, useState } from 'react';
+import { crudPermissions, getLoggedInUser } from '../../utils/common';
+
+const CrudPermissionwithRBAC = <P extends object>(
+  WrappedComponent: React.ComponentType<P>,
+  permission: string
+) => {
+  const WithRBAC: React.FC<P> = (props) => {
+    const userRole = getLoggedInUser().userType;
+    const [isDisabled, setIsDisabled] = useState(true);
+    useEffect(() => {
```

Comment:
Here also I didn't get why we are using useEffect?

Reply:
Resolved